packages/plugins/RedPacket/src/SiteAdaptor/FireflyRedpacketConfirmDialog.tsx

```diff
@@ -0,0 +1,451 @@
+    const { chainId, account } = useChainContext<NetworkPluginID.PLUGIN_EVM>()
+    const { data: ensName } = useReverseAddress(NetworkPluginID.PLUGIN_EVM, account, true)
+    const { data: lensOwnerENS } = useReverseAddress(NetworkPluginID.PLUGIN_EVM, currentLensProfile?.ownedBy, true)
+    const { data: farcasterOwnerENS } = useReverseAddress(
+        NetworkPluginID.PLUGIN_EVM,
+        currentFarcasterProfile?.ownedBy,
+        true,
+    )
+    const { classes, theme } = useStyles()
+    const [currentAccount, setCurrentAccount] = useState(
+        currentLensProfile?.handle || currentFarcasterProfile?.handle || ensName || account,
+    )
+    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
+    const [imageLoading, setImageLoading] = useState(true)
+
+    const { data: price } = useFungibleTokenPrice(NetworkPluginID.PLUGIN_EVM, settings?.token?.address)
+
+    const amount = useMemo(
+        () => leftShift(settings?.total || 0, settings?.token?.decimals),
+        [settings.total, settings.token?.decimals],
+    )
+
+    const { value: themes } = useAsync(async () => {
+        if (!currentAccount) return EMPTY_LIST
+        return FireflyRedPacket.getThemeSettings(
+            currentAccount,
+            settings.total,
+            'fungible',
+            settings.token?.symbol,
+            settings.token?.decimals,
+        )
+    }, [currentAccount, settings.total, settings.token])
+
+    const { state, currentIndex, prev, next } = useStateList<FireflyRedPacketAPI.ThemeSettings | undefined>(themes)
+
+    const isFirst = currentIndex === 0
+    const isLatest = themes?.length && currentIndex === themes.length - 1
+
+    const accounts = useMemo(() => {
+        const { currentFarcasterProfile, currentLensProfile } = fireflyContext
+        return uniqBy(
+            compact([
+                currentLensProfile ?
+                    { icon: <Icons.Lens size={24} />, displayName: currentLensProfile.handle }
+                :   undefined,
+                currentFarcasterProfile ?
+                    { icon: <Icons.Farcaster size={24} />, displayName: currentFarcasterProfile.handle }
+                :   undefined,
+                currentLensProfile?.ownedBy ?
+                    { icon: <Icons.MaskWallet size={24} />, displayName: lensOwnerENS || currentLensProfile.address }
+                :   undefined,
+                currentFarcasterProfile?.ownedBy ?
+                    {
+                        icon: <Icons.MaskWallet size={24} />,
+                        displayName: farcasterOwnerENS || currentFarcasterProfile.ownedBy,
+                    }
+                :   undefined,
+                account ? { icon: <Icons.MaskWallet size={24} />, displayName: ensName || account } : undefined,
+            ]),
+            (x) => x.displayName?.toLowerCase(),
+        )
+    }, [account, fireflyContext, farcasterOwnerENS, lensOwnerENS, ensName])
+
+    // image loading
+    useEffect(() => {
```

Comment:
Use `useAsync`.

Reply:
No, there is nothing asynchronous in this effect. It's just registering to listen for events